source/AsyncRgbLedSimulationDataGenerator.cpp

```diff
@@ -15,57 +19,97 @@ AsyncRgbLedSimulationDataGenerator::~AsyncRgbLedSimulationDataGenerator()
 
 void AsyncRgbLedSimulationDataGenerator::Initialize( U32 simulation_sample_rate, AsyncRgbLedAnalyzerSettings* settings )
 {
+	// Initialize the random number generator with a literal seed to obtain repeatability
+    // Change this for srand(time(NULL)) for "truly" random sequences
+    // NOTICE rand() an srand() are *not* thread safe
+    srand( 42 );
+
 	mSimulationSampleRateHz = simulation_sample_rate;
 	mSettings = settings;
 
-	mSerialSimulationData.SetChannel( mSettings->mInputChannel );
-	mSerialSimulationData.SetSampleRate( simulation_sample_rate );
-	mSerialSimulationData.SetInitialBitState( BIT_HIGH );
+	mMaximumChannelValue = (1 << mSettings->BitSize()) - 1;
+
+    // TODO pass in the analyzer and call GetMinimumSampleRate?
+    mClockGenerator.Init( 12 * 1000000, mSimulationSampleRateHz );
+
+	mLEDSimulationData.SetChannel( mSettings->mInputChannel );
+	mLEDSimulationData.SetSampleRate( simulation_sample_rate );
+	mLEDSimulationData.SetInitialBitState( BIT_HIGH );
```

Comment:
I recommend we set the initial state low. elsewhere I'll mention that we should also remove the high pulse at the end of the data burst. The simulation should just begin as if it was already in a reset pulse.

Reply:
I am slightly confused by this, since in your earlier changes you set the initial bit state to low as well. My reading of the datasheets is that the the line sits high 'normally' and is pulled low to trigger a reset - that's also why I added the 'return to high' in the simulation after a data bit.

I guess your reading is that the line sits in the RESET state all the time, except when data is being sent, and I think what you wrote about changes to ReadBit might confirm that. Not a problem, it's just not what I understood from the timing diagrams in the data-sheets.

Reply:
In the implementations I've seen, I haven't seen the line left high. It definitely could be left high, especially if the controller was extra concerned with the exact time the reset event occurs. For our simulation, let's leave it low all the time though, to better match the implementations I've already seen.